src/setup-beam.js

```diff
@@ -560,38 +559,50 @@ function getRunnerOSVersion() {
   return containerFromEnvImageOS
 }
 
-async function get(url0, pageIdxs) {
-  async function getPage(pageIdx) {
-    const url = new URL(url0)
-    const headers = {}
-    const GithubToken = getInput('github-token', false)
-    if (GithubToken && url.host === 'api.github.com') {
-      headers.authorization = `Bearer ${GithubToken}`
-    }
+async function getUrlResponse(url, headers, attempt = 1) {
+  try {
+    const response = await fetch(url, { headers })
```

Comment:
Fetch will throw an error on timeout, so that's why we are wrapping. Open to other options / ideas to make this cleaner.

Reply:
Try-catch should be Ok here. This is not a huge script, and the handling is isolated.